### PR TITLE
[Accessibility] Add endpoint title to the document title

### LINKED ISSUE #43632: Accessibility: Page title and H1 don't accurately reflect the page content in the My Account sub pages
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

As users navigate through the My Account subpages (Dashboard, Orders, Addresses, Payment Methods, Account Details, Edit Address, etc.), the title of the page and the H1 on the page never changes, though the URL does. It always says "My Account" and never accurately reflects which subpage users are viewing.

Web Content Accessibility Guidelines (WCAG) requires that that [pages have accurate titles](https://www.w3.org/WAI/WCAG21/Understanding/page-titled.html) and [fails pages if the title doesn't identify the contents.](https://www.w3.org/WAI/WCAG21/Techniques/failures/F25) It also required that [headings accurately describe the content.](https://www.w3.org/WAI/WCAG22/Understanding/headings-and-labels.html)

### Expected behavior

When navigating through the My Account pages, as the URL changes, the page title and H1 should change to reflect the contents of the sub-page that the user is viewing.

### Actual behavior

Described above.

### Steps to reproduce

Click through the sub pages on the my account page and observe that the H1 and page title (visible in the browser tab) never change.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by Manussakis**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #43632 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to the My Account page on the frontend.
2. Verify the document title is "My account – {Store name}".
    Example:
    image

3. Go to each one of the My account sub-pages (except "Dashboard" and "Log out" ones) and verify the document title changes to "{Sub-page} – {Store name}". 
    Example
    - "Orders – {Store name}"
    - "Download – {Store name}"
    - "Addresses – {Store name}"
    - "Account details – {Store name}"

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/includes/wc-page-functions.php [hunk 1]
```diff
@@ -33,6 +33,36 @@ function wc_page_endpoint_title( $title ) {
...
+		$endpoint_title = WC()->query->get_endpoint_title( $endpoint, $action );
+		$title['title'] = $endpoint_title ? $endpoint_title : $title['title'];
+
+		remove_filter( 'document_title_parts', 'wc_page_endpoint_document_title_parts' );
+	}
+
+	return $title;
+}
+
+add_filter( 'document_title_parts', 'wc_page_endpoint_document_title_parts' );
```
Comment (Manussakis) [inline plugins/woocommerce/includes/wc-page-functions.php:64]: document_title_parts filter reference: https://developer.wordpress.org/reference/hooks/document_title_parts/

**Comment by Manussakis** [comment]
@dkotter This PR is ready for code review. Thanks!

**Review by dkotter — APPROVED** [review]
@ankitguptaindia @qasumitbagthariya This is ready for QA

**Comment by ankitguptaindia** [comment]
**QA/Test Report-** Needs Remediation ⚠️ 
------------

@Manussakis Thanks for the PR!!
The changes work as expected for default tabs coming from WooCommerce Core plugins like 'Orders', 'Downloads', 'Addresses,' etc. However, when tabs are generated by other Woo extensions like Box Office, Bookings, etc., those tabs do not appear in the page title, even though they have the same markup and style as the core tabs.

Please refer to the screenshots below:


 **Default Tabs**

Screenshot 2024-08-23 at 9 13 06 PM


**Woo extensions tabs**

Screenshot 2024-08-23 at 9 17 01 PM

**Comment by Manussakis** [comment]
@ankitguptaindia Thanks for testing it so thoroughly!

The reason the "Booking" subpage name is not on the browser tab is that the WooCommerce Bookings plugin didn't implement it as an endpoint URL like WooCommerce Core did for "Orders", "Downloads", "Addresses", and "Account details" pages.

We can't tackle this at the WooCommerce Core level since it's not an issue related to its logic. Other plugins may have different logic to implement their specific pages which falls outside WooCommerce Core control.

An alternative is to create an issue on the WooCommerce Bookings plugin when this feature is released.

Any thoughts?

cc: @dkotter

**Comment by ankitguptaindia** [comment]
Thanks, @Manussakis for the detailed explanation. I will discuss this with team and create new issue accordingly.

**Comment by ankitguptaindia** [comment]
**QA/Test Report-** ✅ 
------------

**Testing Environment -** 

<Details>

* WordPress: 6.6.1
* Theme: 
**Storefront Version: 4.6.0  
**Twenty Twenty-Four Version: 1.2
* WooCommerce - Development version with this fix-specific branch.  
* PHP: 8.3
* Web Server: Nginx
* Browser: Chrome - Version 127
* OS: macOS Sonoma 14.6



</Details>


**Test Results -** 

The changes work as intended. Testing was conducted with both the Storefront theme and the Twenty Twenty-Four theme.
 

**Functional Demo /  Screencast -** 


https://github.com/user-attachments/assets/fd7f7d46-f0ac-47ae-8fe3-06198e865e9e

**Comment by ankitguptaindia** [comment]
@Manussakis Please update PR title and status for the next step.

**Comment by Manussakis** [comment]
@vikrampm1 This PR is ready for Woo review.

**Comment by vikrampm1** [comment]
@woocommerce/rubik this PR is ready for your review!

**Review by alexflorisca — APPROVED** [review]
Tested this from Rubik's side, all looks good, thanks for working on this. Appreciate the explanation about not being able to set the title for third party plugins that add tabs to this interface. 

Could we maybe reach out to the main extensions (Subscriptions, Bookings, etc) and create issues for them to do the same?

**Comment by ankitguptaindia** [comment]
>Could we maybe reach out to the main extensions (Subscriptions, Bookings, etc) and create issues for them to do the same? 

@alexflorisca thanks for your input here, this is on my to-do list and I will create issues in respective repos (Booking, Box office and now Subscription too) soon.

**Comment by ankitguptaindia** [comment]
Update for the issues shared here: https://github.com/woocommerce/woocommerce/pull/50900#issuecomment-2307384219 created issues for fixes in respective extensions. 

- Booking: https://github.com/woocommerce/woocommerce-bookings/issues/3859

- EU-VAT: https://github.com/woocommerce/woocommerce-eu-vat-number/issues/448

- Box Office: https://github.com/woocommerce/woocommerce-box-office/issues/546

- Subscription: Working expected with Subscription extension. ✅ 
Screenshot 2024-08-29 at 7 48 23 PM

**Comment by jeffpaul** [comment]
@alexflorisca we'll track those downstream issues separately (see cross-links to this PR), but otherwise looks like this PR is good to merge for Woo core?

**Pull request merged**